Streaming lake ingestion private preview user's guide

eickler opened the pull request:
This is a long-running PR maintained as part of the private/public preview process and made available to customers on an informational base. Comments are welcome.

eickler committed:
docs: User's guide draft for streaming lake ingestion

eickler committed:
docs: Added draft SLA

afri-c8y commented on content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md:
```diff
@@ -0,0 +1,102 @@
+
+### Schema limit configuration
+
+{{< product-c8y-iot >}} provides a feature flag to disable hard schema limits. If you disable these limits, data sent to the tenant is:
```
This sounds like all data will be binned if the flag is set.
Should rather be:
_If you disable these limits, data sent to the tenant **which violates the limits,** is:_

afri-c8y commented on content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md:
```diff
@@ -0,0 +1,102 @@
+
+Where:
+* Δt_writer = Timestamp when the batch is written to the data lake − Publish time of the data at the Pulsar CDC topic
+* Δt_appender = Timestamp when the batch is committed to the Iceberg table − Publish time of the commit data at the pending commits topic
```
Customers might ask about the latency between "Timestamp when the batch is written to the data lake" and "Publish time of the commit data at the pending commits topic".

sourajitsaha17 replied:
Actually I added this section as part of a data freshness calculation (just to internally document it). 
Perhaps this part doesn't fit in the public doc as the writer, appender split is more internal arch details?

eickler replied:
Yeah, I was wondering about the same.

afri-c8y commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+
+Device- or customer-specific data in the form of "[fragments](/concepts/domain-model/#fragments)" resides in separate, dedicated tables within the corresponding folders. For instance, if you send a measurement of type `c8y_EngineMetric`, it will be stored in a table named `c8y_EngineMetric` inside the measurements folder.
+
+To simplify querying the current state of your assets, a consolidated representation of that state is available in "latest tables". The `inventory_latest` and `alarms_latest` tables show the most recent, up-to-date state of your inventory and alarms, respectively, without you needing to reconstruct it from the historical logs.
```
Not present in current preview.

eickler replied:
Yes, do you have a docs issue for updating the folder structure/naming? :-)

afri-c8y replied:
I was rather thinking of adding a note here that it's not available yet in the first place.

afri-c8y commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+{{< /c8y-admon-info >}}
+
+{{< c8y-admon-caution >}}
+The service currently does not support certain combinations of type conflicts with lists. It moves conflicting lists to the trash table.
```
This is the first mention of "trash table". We only use "binning" in other places.

rahj-c8y replied:
`trash` table is elaborated under "Binning" section already. We can highlight `trash` word here and comment as see below/link to "Binning" section.

eickler committed:
docs: Initial screenshot of lake structure

jaro-herod commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+
+
+**Table: inventory.c8y_position**
+| eventType             | id    | time                     | type    | alt | lng     | lat       |
```
```suggestion
| eventType             | source    | time                     | type    | alt | lng     | lat       |
```
We use source also at fragments

jaro-herod commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+| 47635 | 2025-08-20T13:41:39.678Z | sb_nano | Tracker #1 | \[c8y_IsDevice, c8y_Position\] | 10           |
+
+**Table: inventory_latest.c8y_position**
+| id    | time                     | type    | alt | lng     | lat       |
```
```suggestion
| source    | time                     | type    | alt | lng     | lat       |
```

jaro-herod commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+| MANAGED_OBJECT_UPDATE | 47635 | 2025-08-20T13:45:20.002Z | sb_nano | Tracker #1 | \[c8y_IsDevice, c8y_Position\] | 10           |
+
+**Table: inventory.c8y_position**
+| eventType             | id    | time                     | type    | alt | lng     | lat       |
```
```suggestion
| eventType             | source    | time                     | type    | alt | lng     | lat       |
```

jaro-herod commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+| 47635 | 2025-08-20T13:45:20.002Z | sb_nano | Tracker #1 | \[c8y_IsDevice, c8y_Position\] | 10           |
+
+**Table: inventory_latest.c8y_position**
+| id    | time                     | type    | alt | lng    | lat    |
```
```suggestion
| source    | time                     | type    | alt | lng    | lat    |
```

jaro-herod commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+| MANAGED_OBJECT_DELETE | 47635 | 2025-08-20T13:50:20.002Z | sb_nano | Tracker #1 | \[c8y_IsDevice, c8y_Position\] | 10           |
+
+**Table: inventory.c8y_position**
+| eventType             | id    | time                     | type    | alt | lng     | lat       |
```
```suggestion
| eventType             | source    | time                     | type    | alt | lng     | lat       |
```

jaro-herod commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+* By connecting query tools such as BI tools and database browsers through [JDBC or ODBC](/datahub/setting-up-datahub/#dremio-api-user).
+* By using the secure and high performance [querying APIs](https://cumulocity.com/api/datahub/) from applications.
+
+### Understanding the data lake structure
```
We should also describe out namespace structure. 
For now we have namespace with tenant id but that may/will be subject to change

jaro-herod commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+
+Device- or customer-specific data in the form of "[fragments](/concepts/domain-model/#fragments)" resides in separate, dedicated tables within the corresponding folders. For instance, if you send a measurement of type `c8y_EngineMetric`, it will be stored in a table named `c8y_EngineMetric` inside the measurements folder.
+
+To simplify querying the current state of your assets, a consolidated representation of that state is available in "latest tables". The `inventory_latest` and `alarms_latest` tables show the most recent, up-to-date state of your inventory and alarms, respectively, without you needing to reconstruct it from the historical logs.
```
current convention is that cdc and latest tables are placed in separate  namespaces

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/overview.md:
```diff
@@ -0,0 +1,35 @@
+---
+
+{{< c8y-admon-preview >}}
+Streaming Lake Ingestion is currently in Private Preview for selected customers. It is currently continuously being developed, hence features described in the documentation may not be fully available or differ from the documentation text. For more information on Private Previews, see ["Previewing features"](/service-terms/previews/#private-preview).
```
```suggestion
Streaming Lake Ingestion is currently in Private Preview for selected customers. It is continuously being developed, hence features described in the documentation may not be fully available or differ from the documentation text. For more information on Private Previews, see [Previewing features](/service-terms/previews/#private-preview).
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+layout: redirect
+---
+
+Streaming Lake Ingestion is an optional service in {{< product-c8y-iot >}}. To use the service, subscribe to it. After subscription, the service automatically stores all newly incoming data in the lake.
```
Should we add a description or link here on how to subscribe?

eickler replied:
AFAIK there is no self-service subscription yet. I'll change it to contact support.

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+Streaming Lake Ingestion is an optional service in {{< product-c8y-iot >}}. To use the service, subscribe to it. After subscription, the service automatically stores all newly incoming data in the lake.
+
+{{< c8y-admon-info >}}
+On subscription, master data is synchronized with the operational store. This synchronization may take a while to complete. For more information, see "[Monitoring the data lake storage](#monitoring-the-data-lake-storage)".
```
```suggestion
On subscription, master data is synchronized with the operational store. This synchronization may take a while to complete. For more information, see [Monitoring the data lake storage](#monitoring-the-data-lake-storage).
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+{{< c8y-admon-info >}}
+On subscription, master data is synchronized with the operational store. This synchronization may take a while to complete. For more information, see "[Monitoring the data lake storage](#monitoring-the-data-lake-storage)".
+
+The service stores only new data incoming after subscription. It does not automatically move data stored in the {{< product-c8y-iot >}} operational store before subscription. For more information, see "[Migrating to data lake storage](#migrating-to-data-lake-storage)".
```
```suggestion
The service stores only new data incoming after subscription. It does not automatically move data stored in the {{< product-c8y-iot >}} operational store before subscription. For more information, see [Migrating to data lake storage](#migrating-to-data-lake-storage).
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+
+The service stores only new data incoming after subscription. It does not automatically move data stored in the {{< product-c8y-iot >}} operational store before subscription. For more information, see "[Migrating to data lake storage](#migrating-to-data-lake-storage)".
+
+*This feature is not part of the current private preview release.*
```
```suggestion
*This feature is not part of the current Private Preview release.*
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+{{< /c8y-admon-info >}}
+
+
+### Analyzing lake data from Cockpit
```
```suggestion
### Analyzing lake data from Cockpit (#analyzing-lake-data-from-cockpit)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+
+### Analyzing lake data from Cockpit
+
+Use {{< product-c8y-iot >}} Cockpit to access data stored in the {{< product-c8y-iot >}} data lake. Log in to the application, then navigate to the data explorer and select *data lake*. When you enable data lake, the application sources data from the data lake instead of the operational store. You can visualize data over longer time ranges than with the operational store.
```
```suggestion
Use the {{< product-c8y-iot >}} Cockpit application to access data stored in the {{< product-c8y-iot >}} data lake. Log in to the application, then navigate to the data explorer and select **Data lake**. When you enable data lake, the application sources data from the data lake instead of the operational store. You can visualize data over longer time ranges than with the operational store.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+{{< c8y-admon-info >}}
+Data arrives in the lake with a small delay. If you need to see the latest data, switch to the operational store.
+
+*This feature is not part of the current private preview release.*
```
```suggestion
*This feature is not part of the current Private Preview release.*
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+{{< /c8y-admon-info >}}
+
+
+### Analyzing lake data using SQL
```
```suggestion
### Analyzing lake data using SQL (#analyzing-lake-data-using-sql)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+
+Within these folders, tables store the incoming IoT data. For example, the `alarm` table contains a complete, historical log of all alarm changes, while the `inventory` table captures all modifications made to your device master data over time.
+
+Device- or customer-specific data in the form of "[fragments](/concepts/domain-model/#fragments)" resides in separate, dedicated tables within the corresponding folders. For instance, if you send a measurement of type `c8y_EngineMetric`, it will be stored in a table named `c8y_EngineMetric` inside the measurements folder.
```
```suggestion
Device- or customer-specific data in the form of [fragments](/concepts/domain-model/#fragments) resides in separate, dedicated tables within the corresponding folders. For instance, if you send a measurement of type `c8y_EngineMetric`, it will be stored in a table named `c8y_EngineMetric` inside the measurements folder.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+
+To understand how the service transfers data to the data lake, the following sections discuss the general table structure for each {{< product-c8y-iot >}} domain model class.
+
+#### General table structure
```
```suggestion
#### General table structure (#general-table-structure)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+* `type`: The type property of the device, measurement and so on.
+
+
+#### Inventory
```
```suggestion
#### Inventory (#inventory)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+| --------------------- | ----- | ------------------------ | ------- | --- | ------- | --------- |
+| MANAGED_OBJECT_CREATE | 47635 | 2025-08-20T13:41:39.678Z | sb_nano | 67  | 6.15173 | 51.211977 |
+
+The columns of the `c8y_position` table reflect the "c8y_Position" fragment and are set as follows:
```
```suggestion
The columns of the `c8y_position` table reflect the "`c8y_Position` fragment and are set as follows:
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+| 47635 | 2025-08-20T13:41:39.678Z | sb_nano | 67  | 6.15173 | 51.211977 |
+
+{{< c8y-admon-info >}}
+"Fragments" such as "c8y\_IsDevice" are only represented in the "fragments" property of the `inventory` tables.
```
```suggestion
Fragments such as `c8y\_IsDevice` are only represented in the "fragments" property of the `inventory` tables.
```

BeateRixen replied:
Shouldn´t it be `c8y_IsDevice` (without /)?

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+| MANAGED_OBJECT_UPDATE | 47635 | 2025-08-20T13:45:20.002Z | sb_nano | 69  | 6.3213  | 50.425    |
+| MANAGED_OBJECT_DELETE | 47635 | 2025-08-20T13:50:20.002Z | sb_nano | 69  | 6.3213  | 50.425    |
+
+#### Alarms
```
```suggestion
#### Alarms (#alarms)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+| 47635  | 2025-08-19T12:03:27.845Z | c8y_BatteryAlarm | 1     | MAJOR    | ACTIVE | Battery level below 5 percent. |
+
+
+#### Events
```
```suggestion
#### Events (#events)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+As events usually record specific occurrences or state changes for a device at a single point in time and are distinct, the service does not create a "latest" version.
+
+
+#### Measurements
```
```suggestion
#### Measurements (#measurements)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+
+There is one important difference for measurements compared to other data types: There is no `measurement.measurement` table. All relevant data is present in the fragment tables.
+
+This example demonstrates how the service handles nested properties: Like the {{< product-c8y-iot >}} APIs and its operational store, Apache Iceberg supports nested data. Just like with the above JSON example, the property `voltage` is a nested structure containing two properties `value` and `unit`. You can query the contained properties from SQL using path expressions such as `"voltage"."value"` ...
```
```suggestion
This example demonstrates how the service handles nested properties: Like the {{< product-c8y-iot >}} APIs and its operational store, Apache Iceberg supports nested data. Just like with the above JSON example, the property `voltage` is a nested structure containing two properties `value` and `unit`. You can query the contained properties from SQL using path expressions such as `"voltage"."value"`. For more information on querying the data from SQL, see [Example queries](#example-queries).
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+
+This example demonstrates how the service handles nested properties: Like the {{< product-c8y-iot >}} APIs and its operational store, Apache Iceberg supports nested data. Just like with the above JSON example, the property `voltage` is a nested structure containing two properties `value` and `unit`. You can query the contained properties from SQL using path expressions such as `"voltage"."value"` ...
+
+#### Operations
```
```suggestion
#### Operations (#operations)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+For more information related to Iceberg data types, please refer to the [Iceberg specification](https://iceberg.apache.org/spec/#semi-structured-types). Note that there are various [structural limits in Iceberg](/service-terms/quotas/).
+{{< /c8y-admon-info >}}
+
+#### Schema evolution
```
```suggestion
#### Schema evolution (#schema-evolution)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+* When the service discovers a new property, it creates a column to store the data in the associated table.
+
+{{< c8y-admon-info >}}
+Currently, the service does not delete tables and columns when they are not in use anymore. For more information, see Section "[Managing schema evolution](#managing-schema-evolution)".
```
```suggestion
Currently, the service does not delete tables and columns when they are not in use anymore. For more information, see [Managing schema evolution](#managing-schema-evolution).
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+Currently, the service does not delete tables and columns when they are not in use anymore. For more information, see Section "[Managing schema evolution](#managing-schema-evolution)".
+{{< /c8y-admon-info >}}
+
+#### Handling of schema conflicts
```
```suggestion
#### Handling of schema conflicts (#handling-of-schema-conflicts)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+{{< /c8y-admon-caution >}}
+
+
+#### Naming
```
```suggestion
#### Naming (#naming)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+This is a reliability measure. The Iceberg specification considers tables and column names as case sensitive. However, upstream query engines do not generally support case sensitivity and may not be able to query such tables.
+{{< /c8y-admon-info >}}
+
+#### Flat views
```
```suggestion
#### Flat views (#flat-views)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,572 @@
+For legacy analytics tools, {{< company-c8y >}} plans to offer Iceberg views with a flat representation of the Iceberg tables. This is currently under implementation.
+{{< /c8y-admon-info >}}
+
+#### Binning
```
```suggestion
#### Binning (#binning)
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/monitoring.md:
```diff
@@ -0,0 +1,43 @@
+In addition to diagnostic tools, {{< company-c8y >}} provides a [service definition](service-definition.md) that outlines key quality objectives.
+
+{{< c8y-admon-preview >}}
+*This feature is not part of the current private preview release.*
```
```suggestion
*This feature is not part of the current Private Preview release.*
```

rahj-c8y committed:
docs: user guide update with description of reporting alarms and events upon schema violation/evolution

rahj-c8y committed:
docs: add missing heading id and minor changes

eickler committed:
docs: Note about delete notifications, fix of SLA

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Jarek Heród <jaroslaw.herod@cumulocity.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Jarek Heród <jaroslaw.herod@cumulocity.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Jarek Heród <jaroslaw.herod@cumulocity.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Jarek Heród <jaroslaw.herod@cumulocity.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Jarek Heród <jaroslaw.herod@cumulocity.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/overview.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/monitoring.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Merge branch 'develop' into feature/streaming-lake-ingestion-priv-prev

eickler committed:
fix: Fixed anchors

eickler committed:
Merge branch 'develop' into feature/streaming-lake-ingestion-priv-prev

eickler committed:
fix: Removed broken link

eickler commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,629 @@
+| 47640  | EVENT_CREATE          | EVENT     | 2025-07-29T15:15:43Z | /event_fragment_2/nestedObject/invalidFieldName###ASD!ASD!@ | event_fragment_2     | ILLEGAL_FIELD_NAME | Illegal field name                        | InRlc3Qi                           |
+
+
+#### Customer-visible operational communication {#customer-visible-operational-communication}
```
@rahj-c8y Please move the section to "Monitoring the data flow" and phrase it as customer documentation rather than as technical specification. 

Please have a quick look at the docs style guide or the copilot-instructions in this repo. Customers are directly addressed with "you" instead of "the customer". 

Please list concretely the events and alarms that you raise, describing the circumstances under which they appear, preferably with one screenshot from the event viewer and alarm UI rather than the JSON. We might add the JSON lateron (but let's review the concrete structure of it first and check if it complies with our general guidelines -- see other PR comment).

riemenschneider committed:
Update Streaming lake ingestion private preview user's guide with internal folder details (#4342)

Added information about the internal folder in Streaming Lake Ingestion.

rahj-c8y committed:
docs: moved Schema change and violation notifications section to monitoring.md; added sample snapshots

rahj-c8y committed:
Merge remote-tracking branch 'origin/feature/streaming-lake-ingestion-priv-prev' into feature/streaming-lake-ingestion-priv-prev

# Conflicts:
#	content/datahub/streaming-lake-ingestion-bundle/using.md

rahj-c8y committed:
docs: resolve merge conflicts

jaro-herod committed:
CDH-5760: Document that schema is defined by the first processed message

jaro-herod committed:
docs: Improve the wording and typos based on feedback

Co-authored-by: Tobias Riemenschneider <tobias.riemenschneider@cumulocity.com>
Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

jaro-herod committed:
CDH-5760: fix syntax

eickler committed:
Update using.md

Moved note to after conflict resolution explanation. @matj-sag do we have somewhere documentation on the messaging service itself and e.g. on ordering topics?

eickler committed:
Improve documentation on message ordering guarantees

Updated link to the Messaging Service documentation for clarity.

flsc-c8y committed:
docs: [CDH-6179] Streaming Lake Ingestion priv prev - BI views feature (#4325)

* docs: [CDH-6170] Initial draft of BI views section

* docs: [CDH-6170] Rephrase title

* docs: [CDH-6170] Lose undesired word

* docs: [CDH-6170] Make table narrower

* docs: [CDH-6170] Formatting improvements

* docs: [CDH-6170] Moved views.md content to using.md and made some changes to wording and formatting

BeateRixen committed:
Apply suggestion from @BeateRixen

eickler committed:
Typo fixes

eickler committed:
Merge pull request #4360 from Cumulocity-IoT/feature/CDH-5760/streaming-lake-ingestion-base-schema

CDH-5760: Document that schema is defined by the first processed message

flsc-c8y committed:
docs: [CDH-6204] Document streaming lake ingestion limits as agreed upon (#4378)

rahj-c8y committed:
docs: updated alarm ui sample screenshot

eickler committed:
Merge branch 'develop' into feature/streaming-lake-ingestion-priv-prev

eickler committed:
docs: Migration to Streaming Lake Ingestion note

eickler committed:
fix: Link fix

afri-c8y committed:
docs: [CDH-6045] Customer documentation update for latest data and table structure changes

afri-c8y committed:
docs: [CDH-6045] Small fixes

afri-c8y committed:
docs: [CDH-6045] Added a TODO item for flat namespace changes

afri-c8y committed:
docs: [CDH-6045] Fixes from CoPilot review (from the original PR on the Iceflow repository)

eickler committed:
Merge branch 'develop' into feature/streaming-lake-ingestion-priv-prev

eickler committed:
docs: Update to latest implementation version

eickler committed:
docs: Typo fixes

eickler committed:
docs: Added limitations on TLP for measurements

flsc-c8y committed:
docs: [CDH-6400] Update docs for writing inventory entries to CDC tables (#4556)

* docs: [CDH-6400] Update docs for writing inventory entries to CDC tables

* docs: [CDH-6400] Rephrase and comment on other types of delete operations

* docs: [CDH-6400] Changes requested by @eicker

eickler committed:
feat: Catalog integration for streaming lake ingestion

eickler committed:
fix: Spelling correction for the usage section of SLI

eickler committed:
feat: Example queries for streaming lake ingestion

eickler committed:
Small clarification about structured column usage in Dremio SQL

eickler committed:
Added FAQ section

flsc-c8y committed:
docs: [CDH-6413] Add limitation for permissions cannot be revoked (#4592)

* docs: [CDH-6413] Add section for integration with query engines

* Revert "docs: [CDH-6413] Add section for integration with query engines"

This reverts commit e4566ab3483198c47b3020bf5a752c340cefa5a6.

* docs: [CDH-6413] Add section to limitations.md to document that permissions cannot be revoked

eickler committed:
docs: Short FAQ addition

eickler committed:
Fixed heading level

eickler committed:
Smaller editorial changes

eickler committed:
Removed data durablity -- add later

flsc-c8y committed:
docs: [CDH-6526] Document setting of tenant option `CDH_ASSET_WHITELIST` (#4802)

* docs: [CDH-6526] Document setting of tenant option `CDH_ASSET_WHITELIST` for Streaming Lake Ingestion

* Update using.md with CDH_ASSET_WHITELIST details

Moved up to the subscription part

---------

Co-authored-by: André Eickler <797483+eickler@users.noreply.github.com>

riemenschneider committed:
docs: [CDH-6494] add self-service catalog credentials section (#4841)

riemenschneider committed:
docs: [CDH-6393] document master data initialization for latest_inventory (#4843)

jaro-herod committed:
docs: [CDH-6747] Document that empty JSON objects/arrays are omitted

Streaming Lake Ingestion now skips empty {} and [] values instead of
coercing them into the target column's type, per Cumulocity-IoT/iceflow#1001.

eickler committed:
docs: Per-tenant limit and sustained load definition for SLI

eickler committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/streaming-lake-ingestion-priv-prev' into feature/streaming-lake-ingestion-priv-prev

jaro-herod committed:
Merge pull request #4875 from Cumulocity-IoT/docs/CDH-6747-omit-empty-json-containers

docs: [CDH-6747] Document that empty JSON objects/arrays are omitted

riemenschneider committed:
docs: clarify first-character naming restriction for Streaming Lake Ingestion (#4892)

* docs: clarify first-character naming restriction for Streaming Lake Ingestion

Documents that the first character of a fragment/property name must be a
letter — digits, underscores, dashes, and other characters are rejected
outright in that position rather than escaped, unlike everywhere else
in the name.

* docs: clarify first-character naming restriction for Streaming Lake Ingestion

Documents that the first character of a fragment/property name must be a
letter — digits, underscores, dashes, and other characters are rejected
outright in that position rather than escaped, unlike everywhere else
in the name.

riemenschneider committed:
docs: fix naming escape example and missing space (#4895)

eickler committed:
Merge branch 'develop' into feature/streaming-lake-ingestion-priv-prev

eickler committed:
Preparation for public preview

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/overview.md:
```diff
@@ -0,0 +1,39 @@
+---
+
+{{< c8y-admon-preview >}}
+Streaming Lake Ingestion is currently in preview. It is continuously being developed. Features described in the documentation may not be fully available in your tenant, or differ from the documentation text. For more information on previews, see [Previewing features](/service-terms/previews/). For limitations of the preview, see [Current preview limitations.](#limitations)
```
```suggestion
Streaming Lake Ingestion is currently in preview. It is continuously being developed. Features described in the documentation may not be fully available in your tenant, or differ from the documentation text. For more information on previews, see [Previewing features](/service-terms/previews/). For limitations of the preview, see [Current preview limitations](#limitations).
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/limitations.md:
```diff
@@ -0,0 +1,18 @@
+* The automated views only expose change data capture tables.
+* Arrays appear as-is in the views.
+* You may see an `internal` folder with tables for internal service information. Do not modify these tables.
+* The service does not offload any Cumulocity-internal inventory entries, events, measurements and alarms.
```
```suggestion
* The service does not offload any {{< product-c8y-iot >}}-internal inventory entries, events, measurements, and alarms.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/migrating.md:
```diff
@@ -0,0 +1,8 @@
+layout: redirect
+---
+
+You can run DataHub scheduled offloading jobs and Streaming Lake Ingestion in parallel. To migrate your postprocessing or reporting to Streaming Lake Ingestion incrementally, subscribe to Streaming Lake Ingestion and port your queries or scripts as needed. [Contact Cumulocity support](https://cumulocity.atlassian.net/servicedesk/customer/portals) for questions on optimizing your data lake pipelin ...
```
```suggestion
You can run DataHub scheduled offloading jobs and Streaming Lake Ingestion in parallel. To migrate your postprocessing or reporting to Streaming Lake Ingestion incrementally, subscribe to Streaming Lake Ingestion and port your queries or scripts as needed. [Contact {{< company-c8y >}} support](https://cumulocity.atlassian.net/servicedesk/customer/portals) for questions on optimizing your data lake pipelines.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/monitoring.md:
```diff
@@ -0,0 +1,38 @@
+
+
+The service automatically updates the data lake schema when the structure of the incoming data changes. You receive notifications
+in the form of Cumulocity events for schema changes and alarms for schema violations. These notifications help you stay informed about changes and be able to address any issues.
```
```suggestion
in the form of {{< product-c8y-iot >}} events for schema changes and alarms for schema violations. These notifications help you stay informed about changes and be able to address any issues.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/faq.md:
```diff
@@ -0,0 +1,38 @@
+* Use Edge or Data Preparation to change data to more user friendly names, execute simple calculations or route data using processing modes.
+* Use Digital Twin Manager to promote data points to asset level.
+* Use Streaming Analytics to create new data in your desired output format.
+* Use Data Broker to transfer data between tenants.
```
```suggestion
* Use data broker to transfer data between tenants.
```

BeateRixen replied:
We do not capitalize data broker. In this list, this might seem inconsistent, but to be consistent with the rest of the documentation, we should change the spelling here.

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/faq.md:
```diff
@@ -0,0 +1,38 @@
+
+### Is DataHub Query (Dremio) required for Streaming Lake Ingestion?
+
+DataHub Query is optional if you want to post-process or query data using own tools. Note that views are currently only generate for engines compatible with Dremio and Spark SQL. While we designed the service to be as compliant as possible with common tool restrictions, we cannot guarantee compliance with all Iceberg tools.
```
```suggestion
DataHub Query is optional if you want to post-process or query data using your own tools. Note that views are currently only generated for engines compatible with Dremio and Spark SQL. While we designed the service to be as compliant as possible with common tool restrictions, we cannot guarantee compliance with all Iceberg tools.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/faq.md:
```diff
@@ -0,0 +1,38 @@
+
+Only changes after subscription to Streaming Lake Ingestion are visible in the lake. If you have recently subscribed and the data was not changed since then, it will not be transferred to the lake.
+
+If the data was changed and it is not visible, check the "Alarms" overview page in the Cockpit application for warnings. You may have breached a limit. In this case, data breaching the limit is insert into the `trash` table.
```
```suggestion
If the data was changed and it is not visible, check the **Alarms** page in the Cockpit application for warnings. You may have breached a limit. In this case, data that breaches the limit is inserted into the `trash` table.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/overview.md:
```diff
@@ -0,0 +1,39 @@
+
+The Streaming Lake Ingestion feature in DataHub bridges the gap between real-time IoT data and large-scale business analytics and artificial intelligence needs. It takes the continuous stream of IoT data flowing into {{< product-c8y-iot >}} and automatically transforms it into a queryable and AI-ready format in your data lake. This fully managed process requires no configuration to start. When en ...
+
+### From live stream to analytics
```
The headings in this file are missing the IDs.

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+{{< c8y-admon-info >}}
+The download may take a while to complete. For more information, see [Monitoring the data lake storage](#monitoring-the-data-lake-storage).
+
+The `latest_inventory` tables are pre-populated with your full current inventory at subscription time. Alarms, events, measurements and operations are only recorded for changes that occur after subscription.
```
```suggestion
The `latest_inventory` tables are pre-populated with your full current inventory at subscription time. Alarms, events, measurements, and operations are only recorded for changes that occur after subscription.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+
+* Through the user interface of the [embedded query engine](/datahub/setting-up-datahub/#dremio-api-user).
+* By using the secure and high performance [querying APIs](https://cumulocity.com/api/datahub/) from applications and by using JDBC or ODBC with query engines and database tools.
+* By connecting applications directly to the Cumulocity Iceberg catalog.
```
```suggestion
* By connecting applications directly to the {{< product-c8y-iot >}} Iceberg catalog.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+* By using the secure and high performance [querying APIs](https://cumulocity.com/api/datahub/) from applications and by using JDBC or ODBC with query engines and database tools.
+* By connecting applications directly to the Cumulocity Iceberg catalog.
+
+#### Using DataHub Query to analyze lake data
```
This and the following heading is missing an ID.

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+
+#### Obtaining Iceberg catalog credentials {#obtaining-iceberg-catalog-credentials}
+
+To connect to the Cumulocity Iceberg catalog directly — for example, from Apache Spark, Databricks, or a custom application — you need OAuth2 client credentials. As a tenant administrator, you can create and manage named catalog principals using the Manager API.
```
```suggestion
To connect to the {{< product-c8y-iot >}} Iceberg catalog directly — for example, from Apache Spark, Databricks, or a custom application — you need OAuth2 client credentials. As a tenant administrator, you can create and manage named catalog principals using the Manager API.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+
+**Prerequisites**
+
+* Your Cumulocity user must have the `ROLE_TENANT_ADMIN` role.
```
```suggestion
* Your {{< product-c8y-iot >}} user must have the ROLE_TENANT_ADMIN permission.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+
+#### Using the Iceberg catalog from Apache Spark
+
+[Apache Spark](https://spark.apache.org/) is a distributed computing framework that seamlessly integrates with the Cumulocity Iceberg catalogs to provide full SQL-based data processing through a standard [Iceberg REST catalog interface](https://iceberg.apache.org/rest-catalog-spec/).
```
```suggestion
[Apache Spark](https://spark.apache.org/) is a distributed computing framework that seamlessly integrates with the {{< product-c8y-iot >}} Iceberg catalogs to provide full SQL-based data processing through a standard [Iceberg REST catalog interface](https://iceberg.apache.org/rest-catalog-spec/).
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+
+[Apache Spark](https://spark.apache.org/) is a distributed computing framework that seamlessly integrates with the Cumulocity Iceberg catalogs to provide full SQL-based data processing through a standard [Iceberg REST catalog interface](https://iceberg.apache.org/rest-catalog-spec/).
+
+Use [OpenID Connect](https://openid.net/developers/how-connect-works/) with a client credentials grant type to authenticate against the Cumulocity Iceberg REST catalog.
```
```suggestion
Use [OpenID Connect](https://openid.net/developers/how-connect-works/) with a client credentials grant type to authenticate against the {{< product-c8y-iot >}} Iceberg REST catalog.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+
+`https://iceberg.cumulocity.com:19120/api/catalog/v1/oauth/tokens`.
+
+Run Spark SQL queries with the Cumulocity Iceberg catalog on an AWS S3 object store using the example below. In the example, replace
```
```suggestion
Run Spark SQL queries with the {{< product-c8y-iot >}} Iceberg catalog on an AWS S3 object store using the example below. In the example, replace
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+While Iceberg is widely supported, the degree of support currently still varies and we cannot guarantee the catalog to be interoperable with all setups and applications.
+{{< /c8y-admon-info >}}
+
+### Understanding the data lake structure
```
Some more headings are missing IDs, please check throughout this document.

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+* Latest data (`latest`): Provides the current representation of incoming inventory data. For example, the `latest_inventory.inventory` table contains a consolidated view on the latest state of your inventory without you needing to reconstruct the state from the historical logs.
+* Views (`view`): Provides views for the change data capture tables. For example, the `view_alarm.alarm` view shows the same content as `cdc_alarm.alarm` with nested properties converted to top-level columns. This makes the data more easily browsable in tools like PowerBI.
+
+For each type of data, folders ("namespaces") are created to mirror the familiar [{{< product-c8y-iot >}} domain model](/concepts/domain-model/) of Cumulocity as shown in the screenshot. You will find folders with names ending in `inventory`, `alarm`, `event`, `measurement` and `operation`. Inside the folders, tables for standard data, device-specific data and customer-specific data are created.  ...
```
```suggestion
For each type of data, folders ("namespaces") are created to mirror the familiar [{{< product-c8y-iot >}} domain model](/concepts/domain-model/) as shown in the screenshot. You will find folders with names ending in `inventory`, `alarm`, `event`, `measurement` and `operation`. Inside the folders, tables for standard data, device-specific data, and customer-specific data are created. Device- or customer-specific data in the form of [fragments](/concepts/domain-model/#fragments) resides in separate, dedicated tables within the corresponding folders. For instance, if you send a measurement with a fragment type `c8y_EngineMetric`, it will be stored in a table named `cdc_measurement.c8y_EngineMetric`.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+
+All tables contain the following column:
+* `id`: The record ID of the device, asset, alarm, event, measurement or operation associated with the change.
+  In the inventory tables, this is the [managed object ID](https://cumulocity.com/docs/concepts/domain-model/#object-identification).
```
```suggestion
  In the inventory tables, this is the [managed object ID](/concepts/domain-model/#object-identification).
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+  In the inventory tables, this is the [managed object ID](https://cumulocity.com/docs/concepts/domain-model/#object-identification).
+
+The following columns are present in some tables, depending on the type of data:
+* `source`: For alarms, events or measurement, this columns contains the [managed object ID](https://cumulocity.com/docs/concepts/domain-model/#object-identification) of the device or asset associated with the change.
```
```suggestion
* `source`: For alarms, events, or measurement, this column contains the [managed object ID](/concepts/domain-model/#object-identification) of the device or asset associated with the change.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+
+The following columns are present in some tables, depending on the type of data:
+* `source`: For alarms, events or measurement, this columns contains the [managed object ID](https://cumulocity.com/docs/concepts/domain-model/#object-identification) of the device or asset associated with the change.
+* `deviceId`: For operations, this columns contains the [managed object ID](https://cumulocity.com/docs/concepts/domain-model/#object-identification) of the device targeted for the operation.
```
```suggestion
* `deviceId`: For operations, this column contains the [managed object ID](/concepts/domain-model/#object-identification) of the device targeted for the operation.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+The following columns are present in some tables, depending on the type of data:
+* `source`: For alarms, events or measurement, this columns contains the [managed object ID](https://cumulocity.com/docs/concepts/domain-model/#object-identification) of the device or asset associated with the change.
+* `deviceId`: For operations, this columns contains the [managed object ID](https://cumulocity.com/docs/concepts/domain-model/#object-identification) of the device targeted for the operation.
+* `time`: The timestamp of the change as sent by the device. This column is present for alarms, events and measurements.
```
```suggestion
* `time`: The timestamp of the change as sent by the device. This column is present for alarms, events, and measurements.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+* `id` contains "47635", the managed object ID assigned to the new device by {{< product-c8y-iot >}}.
+* `lastUpdated` contains "2025-08-20T13:41:39.678Z", the timestamp of the most recent update to the managed object in {{< product-c8y-iot >}}.
+* `name` contains "Tracker #1", the name of the managed object as used in the request.
+* `owner` contains the user who created the managed object
```
```suggestion
* `owner` contains the user who created the managed object.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+* `childDevices` contains a list of IDs of child devices of an asset or device.
+* `someproperty` contains 10. {{< product-c8y-iot >}} creates additional columns in the `inventory` table when there are top-level properties of an atomic type in the managed object (character strings, numbers, booleans).
+
+Additional columns may be visible depending on your use of Cumulocity. For example, once the device has emitted measurements, you will see a column `supportedMeasurements` with a list of the measurement fragments sent by the device.
```
```suggestion
Additional columns may be visible depending on your use of {{< product-c8y-iot >}}. For example, once the device has emitted measurements, you will see a column `supportedMeasurements` with a list of the measurement fragments sent by the device.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+Empty JSON objects (`{}`) and empty arrays (`[]`) carry no data to store. They are omitted from the input and treated the same as an absent or "null" property, rather than being converted to a value of the column's data type.
+
+{{< c8y-admon-info >}}
+For more information related to Iceberg data types, please refer to the [Iceberg specification](https://iceberg.apache.org/spec/#semi-structured-types). Note that there are various [structural limits](#limits-of-streaming-lake-ingestion) that the service implements to ensure that common upstream query engines can interact with the data produced by the service.
```
```suggestion
For more information related to Iceberg data types, refer to the [Iceberg specification](https://iceberg.apache.org/spec/#semi-structured-types). Note that there are various [structural limits](#limits-of-streaming-lake-ingestion) that the service implements to ensure that common upstream query engines can interact with the data produced by the service.
```

BeateRixen commented on content/datahub/streaming-lake-ingestion-bundle/using.md:
```diff
@@ -0,0 +1,992 @@
+
+#### Basic time series queries
+
+To retrieve all events recorded for a specific device within a given timespan, query the `cdc_event.event` table. Note that `time` is a reserved word in Dremio SQL, so it needs to be quoted.
```
```suggestion
To retrieve all events recorded for a specific device within a given timespan, query the `cdc_event.event` table. Note that `time` is a reserved word in Dremio SQL, so it must be quoted.
```

BeateRixen commented on content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md:
```diff
@@ -0,0 +1,77 @@
+---
+
+{{< c8y-admon-info >}}
+Streaming Lake Ingestion is currently in Private Preview. Preview features are not subject to a service-level agreement. The following text is for information purposes only.
```
change to Public Preview?

BeateRixen commented on content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md:
```diff
@@ -0,0 +1,77 @@
+
+This agreement is made between {{< company-c8y >}} ("Provider") and the Customer ("Customer") who uses {{< product-c8y-iot >}} Streaming Lake Ingestion ("Service") for offloading and analyzing Internet of Things ("IoT") data using Provider's cloud instances ("software-as-a-service", "SaaS").
+
+### Service Description
```
```suggestion
### Service description
```

BeateRixen commented on content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md:
```diff
@@ -0,0 +1,77 @@
+
+You are responsible for:
+
+* Monitoring for {{< product-c8y-iot >}} alarms related to data rejection. For details, see ["Monitoring the data flow"](/datahub/streaming-lake-ingestion/#monitoring).
```
```suggestion
* Monitoring for {{< product-c8y-iot >}} alarms related to data rejection. For details, see [Monitoring the data flow](/datahub/streaming-lake-ingestion/#monitoring).
```

BeateRixen commented on content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md:
```diff
@@ -0,0 +1,77 @@
+
+* **Data model performance**: The Service stores data using a standardized tabular schema optimized for write performance and a broad range of analytical use cases. This schema is not optimally performant for all query patterns. For applications requiring maximum query performance, you must implement post-processing to create optimized data models (for example, "gold-layer" tables) or leverage qu ...
+* **Cost profile variations**: The Service balances data timeliness against cost-efficiency of the underlying storage operations (for example, Amazon S3). The internal mechanisms that manage this trade-off (for example, data batching frequency) are part of the Service's evolving software implementation. As the Service is updated and optimized over time, the hyperscaler cost profile of your usage  ...
+* **Storage reclamation delay**: Apache Iceberg retains historical data versions and deleted files for a period of time to ensure data integrity. These files are permanently removed during periodic, automated cleanup cycles. This intentional delay in file deletion affects your total billed storage, as physical storage space is not reclaimed in real-time.
```
```suggestion
* **Storage reclamation delay**: Apache Iceberg retains historical data versions and deleted files for a period of time to ensure data integrity. These files are permanently removed during periodic, automated cleanup cycles. This intentional delay in file deletion affects your total billed storage, as physical storage space is not reclaimed in realtime.
```

BeateRixen commented on content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md:
```diff
@@ -0,0 +1,77 @@
+
+The service quality indicators are defined as follows:
+
+* **Catalog Availability**: The uptime of the Iceberg catalog service, as defined by the [{{< company-c8y >}} service availability terms](/service-terms/service-level/#service-availability).
```
```suggestion
* **Catalog availability**: The uptime of the Iceberg catalog service, as defined by the [{{< company-c8y >}} service availability terms](/service-terms/service-level/#service-availability).
```

BeateRixen commented on content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md:
```diff
@@ -0,0 +1,77 @@
+The service quality indicators are defined as follows:
+
+* **Catalog Availability**: The uptime of the Iceberg catalog service, as defined by the [{{< company-c8y >}} service availability terms](/service-terms/service-level/#service-availability).
+* **Data Freshness**: The time between message arrival in the {{< product-c8y-iot >}} platform (post-preparation) and its availability for query in the data lake, measured over a calendar month.
```
```suggestion
* **Data freshness**: The time between message arrival in the {{< product-c8y-iot >}} platform (post-preparation) and its availability for query in the data lake, measured over a calendar month.
```

eickler committed:
Update content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/service-terms/service-level-bundle/streaming-lake-ingestion-sla.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/faq.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/monitoring.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/migrating.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/limitations.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/overview.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Heading IDs added (through Copilot)

eickler committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/streaming-lake-ingestion-priv-prev' into feature/streaming-lake-ingestion-priv-prev

eickler committed:
Apply suggestion from @BeateRixen

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Apply suggestion from @BeateRixen

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Apply suggestion from @BeateRixen

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Apply suggestion from @BeateRixen

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Clarified object store requirements in the SLA

eickler committed:
Merge branch 'develop' into feature/streaming-lake-ingestion-priv-prev

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

eickler committed:
Update content/datahub/streaming-lake-ingestion-bundle/using.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>
